utils/terrascript_client.py

```diff
@@ -1232,6 +1235,103 @@ def populate_tf_resource_s3_cloudfront(self, resource, namespace_info):
+        # Terraform resource reference:
+        # https://www.terraform.io/docs/providers/aws/r/
+        # cloudwatch_log_group.html
+        values = {
+            'name': identifier,
+            'tags': common_values['tags'],
+            'retention_in_days':
+                self._get_retention_in_days(
+                    common_values, account, identifier)
+        }
+
+        region = common_values['region'] or self.default_regions.get(account)
+        if self._multiregion_account_(account):
+            values['provider'] = 'aws.' + region
+        log_group_tf_resource = aws_cloudwatch_log_group(identifier, **values)
+        tf_resources.append(log_group_tf_resource)
+        output_name = output_prefix + '[log_group_name]'
+        output_value = '${' + log_group_tf_resource.fullname + '.name}'
+        tf_resources.append(output(output_name, value=output_value))
+        output_name = output_prefix + '[aws_region]'
+        tf_resources.append(output(output_name, value=region))
+
+        # iam resources
+        # Terraform resource reference:
+        # https://www.terraform.io/docs/providers/aws/r/iam_access_key.html
+
+        # iam user for log group
+        values = {
+            'name': identifier,
+            'tags': common_values['tags'],
+            'depends_on': [log_group_tf_resource]
+        }
+        user_tf_resource = aws_iam_user(identifier, **values)
+        tf_resources.append(user_tf_resource)
+
+        # iam access key for user
+        tf_resources.extend(
+            self.get_tf_iam_access_key(
+                user_tf_resource, identifier, output_prefix))
+
+        # iam user policy for log group
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Action": [
+                        "logs:CreateLogStream",
+                        "logs:DescribeLogStreams",
+                        "logs:PutLogEvents",
+                        "logs:GetLogEvents"
+                    ],
+                    "Effect": "Allow",
+                    "Resource":
+                        "${" + log_group_tf_resource.fullname + ".arn}:*"
+                }
+            ]
+        }
+        values = {
+            'user': identifier,
+            'name': identifier,
+            'policy': json.dumps(policy, sort_keys=True),
+            'depends_on': [user_tf_resource]
+        }
+        tf_resource = aws_iam_user_policy(identifier, **values)
+        tf_resources.append(tf_resource)
+
+        for tf_resource in tf_resources:
+            self.add_resource(account, tf_resource)
+
+    @staticmethod
+    def _get_retention_in_days(values, account, identifier):
+        default_retention_in_days = 14
+        allowed_retention_in_days = \
+            [1, 3, 5, 7, 14, 30, 60, 90, 120, 150,
+             180, 365, 400, 545, 731, 1827, 3653]
+
+        retention_in_days = \
+            values.get('retention_in_days') or default_retention_in_days
```

Comment:
Would it help to add a check here to make sure the value is a number? What would happen if user provided `14d` instead of `14`? Can we catch this error and show a ERROR in logs?

Reply:
changed logging from warning to error